index.js

```diff
@@ -305,252 +301,89 @@ app.get('/slack', function(req, res) {
-    text: '',
-    updated: null,
-    newlink: '/docs/DIYModule'
-  }
-};
+// All the docs redirects
+var gitbooksBase = 'https://tessel.gitbooks.io/docs/content/';
 
-var sideBarLists = {
-  'API': [],
-  'tutorials': [],
-  'module': []
-};
-Object.keys(rawDocs).forEach(function(item) {
-  var doc = rawDocs[item];
-  if (sideBarLists[doc.type])
-    sideBarLists[doc.type].push({
-      link: item,
-      clean: doc.clean
-    });
+app.get('/docs', function(req, res) {
+  res.redirect(gitbooksBase);
 });
 
-function replaceAll(find, replace, str) {
-  return str.replace(new RegExp(find, 'g'), replace);
-}
+//********* Begin legacy redirects for docs *********//
+// These legacy link redirects keep links posted around the internet functional with new Gitbooks version of docs
 
-var renderer = new marked.Renderer();
-var headingList = [];
-
-renderer.heading = function(text, level) {
-  var escapedText = text.toLowerCase().replace(/[^\w]+/g, '-');
-  var link = "";
-  var divClass = "";
-  if (level <= 2) {
-    if (level == 2) {
-      headingList.push({
-        link: escapedText,
-        text: text
-      });
-    }
-    link = '<i class="fa fa-link fa-1"></i>';
-    divClass = 'class="heading-div"';
-  }
-  return '<div ' + divClass + '><a name="' + escapedText + '" class="docHeader-' + level + '" href="#' +
-    escapedText + '"><span>' + link + '</span></a><h' + level + ' class="cliHeader">' + text + '</h' + level + '></div>';
-};
+var originalBase = '/docs/';
 
-renderer.list = function(body, ordered) {
-  if (ordered) {
-    return '<ol class="list-div">' + body + '</ol>';
-  } else {
-    return '<ul class="list-div">' + body + '</ul>';
-  }
-};
+app.get(originalBase + 'home', function (req, res) {
+  res.redirect(gitbooksBase);
+});
 
-renderer.code = function(code, lang) {
-  if (lang == 'js') {
-    return '<pre><code class="lang-js"><div style="line-height:1.1em;">' + require('highlight.js').highlightAuto(code).value + '</div></code></pre>';
-  } else {
-    return '<code><cli>' + code + '</cli></code>';
-  }
-};
+app.get(originalBase + 'hardwareAPI', function (req, res) {
+  res.redirect(gitbooksBase + 'API/Hardware_API');
+});
 
-app.get('/docs/:slug', function(req, res) {
-  var doc = req.params.slug;
-  if (!rawDocs[doc]) {
-    return res.status(404).send("Sorry page not found");
-  }
-  var forkPath = uri.parse(rawDocs[doc].url).pathname.split('/');
-  var forkUrl = '//github.com/' + forkPath[1] + '/' + forkPath[2];
-  // update on anything older than 1 hour
-  var d = new Date((new Date()) * 1 - 1000 * 3600 * 1);
-  // if any urls have blank code, request it
-  if ((rawDocs[doc].text === '' && rawDocs[doc].url) || (rawDocs[doc].updated < d.valueOf())) {
-    request.get(rawDocs[doc].url, function(err, data) {
-      if (err) {
-        return console.error('could not get', doc, rawDocs[doc].text, err);
-      }
-      // Turn markdown into html
-      headingList = [];
-      var text = marked(data.body, {
```

Comment:
@tcr @jiahuang can't remember which of you I worked with on this originally. Is this the only place we use `marked`/can I remove the dependency?

I think there are other modules we pulled in only to parse docs pages. Is there a good tool to discover which modules are no longer needed (or is CTRL+F standard for this)?